### PR TITLE
docs(Foundation/List): update documentation template

### PR DESCRIPTION
**Pull request by toncabral88**
## Description
https://jirasoftware.catho.com.br/browse/QTM-360

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [ ] Code review
- [ ] Documentation updated

### CODE REVIEW

**Review by MarcosViniciusPC — COMMENTED** [review]
I found the behavior a bit confusing that just clicking on the default example goes to the top of the documentation list and the other examples scroll to their respective examples. What do you think? Is it necessary to keep this list of examples as sub-items of the list?

**Comment by toncabral88** [comment]
> I found the behavior a bit confusing that just clicking on the default example goes to the top of the documentation list and the other examples scroll to their respective examples. What do you think? Is it necessary to keep this list of examples as sub-items of the list?

I don't like this behavior too, but I think this is an issue of storybook or some like that. I'll search if there's an alternative for it.

**Review by rapahaeru — APPROVED** [review]

File: stories/List/List.story.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+```js
+import { List } from '@catho/quantum';
+```
+
+## <a id="api"></a>API
+
+<ArgsTable of={List} />
+
+## <a id="examples"></a>Examples
+
```
Comment (MarcosViniciusPC) [inline stories/List/List.story.mdx:40]:
```suggestion
### Default
```

File: stories/List/List.story.mdx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+<Canvas>
+  <Story story={Default} />
+</Canvas>
+
```
Comment (MarcosViniciusPC) [inline stories/List/List.story.mdx:44]:
```suggestion
### Ordered
```

File: stories/List/List.story.mdx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+<Canvas>
+  <Story story={Ordered} />
+</Canvas>
+
```
Comment (MarcosViniciusPC) [inline stories/List/List.story.mdx:48]:
```suggestion
### Divided
```

File: stories/List/List.story.mdx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+<Canvas>
+  <Story story={Divided} />
+</Canvas>
+
```
Comment (MarcosViniciusPC) [inline stories/List/List.story.mdx:52]:
```suggestion
### HeaderAndSubHeader
```

File: stories/List/List.story.mdx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+<Canvas>
+  <Story story={HeaderAndSubHeader} />
+</Canvas>
+
```
Comment (MarcosViniciusPC) [inline stories/List/List.story.mdx:56]:
```suggestion
### WithIcons
```

**Review by MarcosViniciusPC — APPROVED** [review]
[x] Code Review
[x] Documentation

**Review by MarcosViniciusPC — APPROVED** [review]
[x] Code review  
[x] Documentation

**Review by rapahaeru — APPROVED** [review]

**Pull request merged**